.github/workflows/cypress-tests-runner.yml

```diff
@@ -0,0 +1,175 @@
+name: Run Cypress tests
+
+on: 
+  workflow_dispatch:
+  pull_request:
+  merge_group:
+    types: [checks_requested]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 1
+  CARGO_NET_RETRY: 10
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
+  CONNECTORS: stripe
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+  RUST_MIN_STACK: 8388608
+
+jobs:
+  runner:
+    name: Run Cypress tests
+    runs-on: ubuntu-latest
+
+    services:
+      redis:
+        image: "redis"
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
+      postgres:
+        image: "postgres:14.5"
+        env:
+          POSTGRES_USER: db_user
+          POSTGRES_PASSWORD: db_pass
+          POSTGRES_DB: hyperswitch_db
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v4
+
+      - name: Download Encrypted TOML from S3 and Decrypt
+        if: ${{ ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name)) || (github.event_name == 'merge_group')}}
```

Comment:
Instead of repeating this long `if` condition everywhere, we can set an environment variable (say `SKIP_CYPRESS_TESTS` or `SKIP_TESTS`) based on these conditions, and use this environment variable to skip steps in further steps. This should make it easier for us to tweak the condition on a later day as well.